src/main/java/org/candlepin/subscriptions/task/TaskManager.java

```diff
@@ -81,36 +82,38 @@ public void updateAccountSnapshots(String accountNumber) {
      *
      * @throws TaskManagerException
      */
+    @Transactional
     public void updateSnapshotsForAllAccounts() {
-        List<String> accountList;
-        try {
-            accountList = accountListSource.list();
-        }
-        catch (IOException ioe) {
-            throw new TaskManagerException("Could not list accounts for update snapshot task generation",
-                ioe);
-        }
+        final AtomicInteger count = new AtomicInteger();
+        try (Stream<String> accountList = accountListSource.syncableAccounts()) {
+            int accountBatchSize = appProperties.getAccountBatchSize();
+            log.info("Queuing snapshot production in batches of {}.", accountBatchSize);
 
-        int accountBatchSize = appProperties.getAccountBatchSize();
-        log.info("Queuing snapshot production for {} accounts in batches of {}.", accountList.size(),
-            accountBatchSize);
+            accountList
+                .collect(Collectors.groupingBy(a -> count.getAndIncrement() / accountBatchSize))
+                .values()
+                .forEach(accounts -> {
```

Comment:
I need to find a better way of chunking up this data. As it is right here, the complete list will be read in up front and split accordingly.

@kahowell @awood Any ideas here? I'm not sure that I can accomplish this with the streaming API.

Reply:
Thinking that I might have to use paging to get the result I'm after. Thoughts?

Reply:
@mstead, you could do something similar to the pattern you used in RedHatInsights/rhsm-conduit#117 ?